bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param xfPage the experience fragment page
+     * @param sitePage the content page
+     * @return
+     */
+    private Page getXfLocalizedPage(@NotNull Page xfPage, @NotNull Page sitePage) {
+        Resource siteRoot = resolver.getResource(localizationRoot);
+        if (siteRoot == null) {
+            return null;
+        }
+        String xfLocalizationRoot = StringUtils.replace(localizationRoot, CONTENT_ROOT, EXPERIENCE_FRAGMENTS_ROOT, 1);
+        Resource xfRoot = resolver.getResource(xfLocalizationRoot);
+        if (xfRoot == null) {
+            return null;
+        }
+
+        String xfPageLocalizationString = getLocalizationString(xfPage, xfRoot, localizationDepth); // e.g. us/en
+        String sitePageLocalizationString = getLocalizationString(sitePage, siteRoot, localizationDepth); // e.g. us/es
+        if (StringUtils.isEmpty(xfPageLocalizationString)) {
+            return null;
+        }
+        if (StringUtils.isEmpty(sitePageLocalizationString)) {
+            return null;
+        }
+
+        String xfLocalizationSrcPath = StringUtils.joinWith(PATH_DELIMITER, xfLocalizationRoot, xfPageLocalizationString); // e.g. /content/experience-fragments/my-site/us/en
+        String xfLocalizationDestPath = StringUtils.joinWith(PATH_DELIMITER, xfLocalizationRoot, sitePageLocalizationString); // e.g. /content/experience-fragments/my-site/us/es
+
+        String xfLocalizedPagePath = StringUtils.replace(xfPage.getPath(), xfLocalizationSrcPath, xfLocalizationDestPath, 1);
+        return pageManager.getPage(xfLocalizedPagePath);
+    }
+
+    /**
+     * Returns the localization string of the given page based on the localization root and depth.
+     *
+     * Use case                                  | Page path                         | localizationRoot  | localizationDepth | localizationString (output)
+     * ----------------------------------------  |-----------------------------------|-------------------|-------------------|----------------------------
+     * 1. No localization                        | /content/my-site/my-page          | empty string      | 0                 | empty string
+     * 2. Language localization                  | /content/my-site/en/my-page       | /content/my-site  | 1                 | en
+     * 3. Country-language localization          | /content/my-site/us/en/my-page    | /content/my-site  | 2                 | us/en
+     * 4. Country-language localization (variant)| /content/us/my-site/en/my-page    | /content          | 3                 | us/my-site/en
+     *
+     * @param page the page that contains the localization string
+     * @param root the localization root resource
+     * @param depth the localization depth
+     * @return the localization string
+     */
+    private String getLocalizationString(@NotNull Page page, @NotNull Resource root, int depth) {
```

Comment:
```suggestion
    private String getLocalizationString(@NotNull String pagePath, @NotNull Resource root, int depth) {
```
It's more flexible to just take what is really needed for a method